pkg/proctree/nodes/process.go

```diff
@@ -0,0 +1,230 @@
+package nodes
+
+import "sync"
+
+// Arguments to feed or retrieve a process node
+
+type ProcessInfo struct {
+	Tid       int32
+	NsTid     int32
+	Pid       int32
+	NsPid     int32
+	StartTime uint64
+	ExitTime  uint64
+}
+
+type FileInfo struct {
+	Name      string
+	Path      string
+	Device    uint32
+	Ctime     uint64
+	Inode     uint64
+	InodeMode uint16
+}
+
+type ProcessForkData struct {
+	Parent ProcessInfo
+	Child  ProcessInfo
+}
+
+type ProcessExecData struct {
+	Command     FileInfo
+	Interpreter FileInfo
+	Pid         int32
+	StartTime   uint64
+}
+
+type ProcessExitData struct {
+	ExitedPid int32
+	ExitCode  int64
+	StartTime uint64
```

Comment:
I'm neutral on the type (we don't really use it much in tracee but maybe we should start to do so).
However, clarification of units should definitely be added.

Reply:
Alright. Anything related to feeding the data structures I created for the proc tree, uses uint64 starttime or exittime. They mean amount of nanoseconds since boottime. I have renamed StartTime and ExitTime to StartTimeNS and ExitTimeNS (and they methods that encapsulate access to them). 

I have also created methods that return "time.Time" with a meaningful time we can use, for both (start and exit times). AN example can be seen at:

```go

	// Walk the process tree and create a table row for each process:

	for _, hash := range pt.processes.Keys() { // for each process
		process, ok := pt.processes.Get(hash)
		if !ok {
			continue
		}
		if process.info.HasExited() {
			continue // only running processes
		}

		// create a row for the table
		processFeed := process.GetInfo().GetFeed()
		execName := process.GetExecutable().GetName()
		if len(execName) > 25 {
			execName = execName[:20] + "..."
		}
		tid := fmt.Sprintf("%v", processFeed.Tid)
		pid := fmt.Sprintf("%v", processFeed.Pid)
		ppid := fmt.Sprintf("%v", processFeed.PPid)
		date := process.GetInfo().GetStartTime().Format("2006-01-02 15:04:05")

		// add the row to the table
		unsortedRows = append(unsortedRows,
			[]string{
				ppid, tid, pid, date, execName,
				getListOfChildrenPids(process),
				getListOfThreadsTids(process),
			},
		)
	}
```

At proctree_output.go. The result is:

```
+--------+--------+--------+---------------------+---------------------------+-------------------------+-------------------------+
| PPID   | TID    | PID    | DATE                | CMD                       | CHILDREN                | THREADS                 |
+--------+--------+--------+---------------------+---------------------------+-------------------------+-------------------------+
| 0      | 1      | 1      | 2023-09-07 18:10:11 | systemd                   | 516,288,3533,1254,18... |                         |
| 0      | 2      | 2      | 2023-09-07 18:10:11 | kthreadd                  | 205265,353,360,4,190... |                         |
| 2      | 3      | 3      | 2023-09-07 18:10:11 | rcu_gp                    |                         |                         |
| 2      | 4      | 4      | 2023-09-07 18:10:11 | rcu_par_gp                |                         |                         |
| 2      | 5      | 5      | 2023-09-07 18:10:11 | slub_flushwq              |                         |                         |
| 2      | 6      | 6      | 2023-09-07 18:10:11 | netns                     |                         |                         |
| 2      | 8      | 8      | 2023-09-07 18:10:11 | kworker/0:0H-events_...   |                         |                         |
...
```
As you can see the starttime in the process tree. 

I'll adapt the "time.Time" for start and exit time when re-introducing the changelog feature. Marking this as solved for now.